logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -126,6 +126,10 @@ public long getCurrentByteSize() {
         return this.currentByteSize;
     }
 
+    public long getCurrentPhysicallyPersistedByteSize() {
+        return headPage.getPageIO().getHead() + tailPages.stream().mapToLong((p) -> p.getPageIO().getHead()).sum();
```

Comment:
This strategy seems good. Sum of the head position (byte value offset) of all pages.
👍